openedx/core/djangoapps/user_api/accounts/serializers.py

```diff
@@ -63,6 +68,16 @@ def to_representation(self, user):
         :return: Dict serialized account
         """
         profile = user.profile
+        badges = settings.FEATURES.get('ENABLE_OPENBADGES') or False
+        if 'badges' in self.excluded_fields:
+            # Avoid doing this potentially expensive query-- it will be filtered out anyway.
```

Comment:
If you change this to a Boolean in the profile API, doesn't it satisfy the Web UI's requirements?  You would then make a separate Ajax call to the badges API.

Sent from my iPhone

> On Dec 11, 2015, at 7:44 AM, Kelketek <notifications@github.com> wrote:
> 
> In openedx/core/djangoapps/user_api/accounts/serializers.py:
> 
> > @@ -63,6 +68,16 @@ def to_representation(self, user):
> >          :return: Dict serialized account
> >          """
> >          profile = user.profile
> > +        badges = settings.FEATURES.get('ENABLE_OPENBADGES') or False
> > +        if 'badges' in self.excluded_fields:
> > +            # Avoid doing this potentially expensive query-- it will be filtered out anyway.
> @talbs @antoviaque @nasthagiri The current UX compositions do not have anything for pagination. It will take a while to modify the code to allow this. I don't think we'll have it completed by this week, accordingly, though it should be possible to finish it next week.
> 
> —
> Reply to this email directly or view it on GitHub.
>